### PR TITLE
Replace Listener2 with MultiListener

### LINKED ISSUE #35: Remove `Listener2` mixin
I hate it. If Dart doesn't allow this, let's not fight the language.

IMHO we should strive to be a small, nonbloated, pretty and cute helper lib.

`Listener2` mixin was added in #34.

### PR DESCRIPTION
**Pull request by iasiu**
* Remove Listener2

* Add MultiListener and ListenerDelegate

Closes #35

### CODE REVIEW

File: packages/comms/lib/src/multi_listener.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    listen();
+  }
+
+  @protected
+  @nonVirtual
+  @override
+  void onMessage(T message) => _onMessage(message);
+}
+
+mixin MultiListener {
```
Comment (bartekpacia) [inline packages/comms/lib/src/multi_listener.dart:21]: Some doc comment would be nice :)

**Review by bartekpacia — APPROVED** [review]
💙

File: packages/comms/lib/src/multi_listener.dart (hunk 1 above)
Comment (iasiu) [reply packages/comms/lib/src/multi_listener.dart:21]: it's just experimental, and hope we will be able to move to reflection soon

Comment (iasiu) [reply packages/comms/lib/src/multi_listener.dart:21]: Ok, you convinced me again, will add it

Comment (bartekpacia) [reply packages/comms/lib/src/multi_listener.dart:21]: > reflection

You mean `dart:mirrors`? It's disabled in Flutter. I think you meant static metaprogramming 👅

**Review by bartekpacia — APPROVED** [review]

File: packages/comms/lib/src/multi_listener.dart (hunk 1 above)
Comment (iasiu) [reply packages/comms/lib/src/multi_listener.dart:33]: isn't reflection part of metaprogramming?

Comment (bartekpacia) [reply packages/comms/lib/src/multi_listener.dart:33]: Of _metaprogramming_ – I think yes, but not a part of _static metaprogramming_.

Comment (bartekpacia) [reply packages/comms/lib/src/multi_listener.dart:33]: don't @ me

**Pull request merged**